tests/trx_generator/http_client_async.hpp

```diff
@@ -80,8 +80,10 @@ class session : public std::enable_shared_from_this<session> {
       req_.prepare_payload();
 
       // Look up the domain name
-      resolver_.async_resolve(
-          host, std::to_string(port), [self = this->shared_from_this()](beast::error_code ec, auto res) { self->on_resolve(ec, res); });
+      boost::asio::post( resolver_.get_executor(), [host, port, self=this->shared_from_this()] {
+         self->resolver_.async_resolve(
+                 host, std::to_string(port), [self](beast::error_code ec, auto res) { self->on_resolve(ec, res); });
+      });
```

Comment:
Does this change make a difference? I would have thought that, since we provide an executor when building the resolver, the async functions would already be posted on this executor?

Reply:
This code follows the example of `libs/beast/example/http/client/async/http_client_async.cpp` however I think it would be better to follow https://www.boost.org/doc/libs/1_83_0/libs/beast/example/http/client/crawl/http_crawl.cpp as that is a multi-threaded example. I'll close this PR and create a new one with those changes.